engine/runtime/src/main/java/org/enso/interpreter/runtime/library/dispatch/TypesLibrary.java

```diff
@@ -0,0 +1,73 @@
+package org.enso.interpreter.runtime.library.dispatch;
+
+import com.oracle.truffle.api.library.GenerateLibrary;
+import com.oracle.truffle.api.library.Library;
+import com.oracle.truffle.api.library.LibraryFactory;
+import org.enso.interpreter.runtime.data.Type;
+
+/**
+ * A library used for equipping data structures with Enso-style (fully unapplied) method dispatch.
+ * This is used for all method calls in the language.
+ */
+@GenerateLibrary
+@GenerateLibrary.DefaultExport(DefaultLongExports.class)
+@GenerateLibrary.DefaultExport(DefaultDoubleExports.class)
+@GenerateLibrary.DefaultExport(DefaultBooleanExports.class)
+public abstract class TypesLibrary extends Library {
+
+  /**
+   * Returns a resolved library factory for this library.
+   *
+   * @return a library factory instance
+   */
+  public static LibraryFactory<TypesLibrary> getFactory() {
+    return FACTORY;
+  }
+
+  /**
+   * Returns the uncached instance of this library.
+   *
+   * @return the uncached instance of this library
+   */
+  public static TypesLibrary getUncached() {
+    return FACTORY.getUncached();
+  }
+
+  private static final LibraryFactory<TypesLibrary> FACTORY =
+      LibraryFactory.resolve(TypesLibrary.class);
+
+  /**
+   * Checks if the receiver supports Enso-style method dispatch
+   *
+   * @param receiver the receiver to check
+   * @return whether the receiver supports method dispatch through this library
+   */
+  @GenerateLibrary.Abstract(ifExported = {"getType"})
+  public boolean hasType(Object receiver) {
+    return false;
+  }
+
+  /**
+   * Denotes an object that has special dispatch semantics, that must be handled specially in the
+   * method dispatch logic.
+   *
+   * @param receiver the receiver to check
+   * @return whether the receiver has special dispatch semantics
+   */
+  public boolean hasSpecialDispatch(Object receiver) {
+    return false;
+  }
+
+  /**
+   * Looks up the method definition for the given receiver and symbol
+   *
+   * @param receiver the method call receiver
+   * @param symbol the symbol being dispatched
```

Comment:
Where is this `symbol` symbol?

Reply:
I've sent to it a farm, where it can happily run around with other symbols.